### ISSUE TITLE
Have users verify email upon signing up

### ISSUE THREAD

**Issue by michaelduan8** [ProjectSidewalk/SidewalkWebpage#2056]
Since we will soon have an email system set up for users to reset their password, it makes sense to have users at least verify their emails when they signup. This will prevent issues like users using non-existent emails or making a typo in their email while signing up, which could be dicey in the case that the said user forgets their password.

**Comment by jonfroehlich** [comment]
@misaugstad after five years since this ticket was opened, do we still tink we need this? I could imagine it would be important if we had issues with spam account creation...

**Comment by misaugstad** [comment]
Duplicate of #4491, which has the fuller writeup (reuse the `AuthToken` machinery from password reset, unverified flag + nag banner, soft vs. hard gating, resend link) and depends on #4493 for reliable transactional email. Closing in favor of that one.
